docs/core_abstraction/node.md

```
@@ -1,338 +1,654 @@
-# Node: The Fundamental Building Block
+# Nodes
 
-A **Node** is the smallest reusable unit in BrainyFlow. Each Node follows a 3-step lifecycle that enforces the principle of separation of concerns.
+Nodes are the fundamental building blocks in BrainyFlow. Each node performs a specific task within your workflow, processing data and optionally triggering downstream nodes.
 
 ## Node Lifecycle
 
 <div align="center">
   <img src="https://github.com/zvictor/brainyflow/raw/main/.github/media/node.jpg" width="400"/>
```

markdownlint-cli2 (0.17.2)

Images should have alternate text (alt text)
null

(MD045, no-alt-text)